crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -76,17 +78,79 @@ impl DynamicScene {
         scene
     }
 
+    /// Create a new dynamic scene from a given world at a given root.
+    pub fn from_world_at_root(
+        world: &World,
+        root: Entity,
+        type_registry: &AppTypeRegistry,
+    ) -> Self {
+        let mut scene = DynamicScene::default();
+        let type_registry = type_registry.read();
+
+        // Extract all entities that are a descendent of root
+        let mut entities_of_interest = Vec::new();
+        let mut entities_to_process = vec![root];
+        while let Some(entity) = entities_to_process.pop() {
+            entities_of_interest.push(entity);
+            if let Some(children) = world.entity(entity).get::<Children>() {
+                entities_to_process.extend(children.iter());
+            }
+        }
+
+        for archetype in world.archetypes().iter() {
+            let entities_offset = scene.entities.len();
+
+            // Create a new dynamic entity for each entity of the given archetype
+            // and insert it into the dynamic scene.
+            for entity in archetype
+                .entities()
+                .iter()
+                .filter(|entity| entities_of_interest.contains(entity))
```

Comment:
Oof, I really don't like that this is O(n). Is there no O(1) alternative?

Reply:
changed the code to iterate only on interesting entities